lib/app/features/nostr/providers/nostr_notifier.c.dart

```diff
@@ -72,9 +77,43 @@ class NostrNotifier extends _$NostrNotifier {
     RequestMessage requestMessage, {
     ActionSource actionSource = const ActionSourceCurrentUser(),
   }) async* {
-    //TODO: handle NOTICE
     final relay = await _getRelay(actionSource);
-    yield* nd.requestEvents(requestMessage, relay);
+
+    final dislikedRelaysUrls = <String>[];
+    StreamSubscription<RelayMessage>? subscription;
+    final streamController = StreamController<EventMessage>(onCancel: () => subscription?.cancel());
+
+    Future<void> handleRetry() => withRetry(
+          () async {
+            final retryRelay = await _getRelay(actionSource, dislikedUrls: dislikedRelaysUrls);
+            final retryEventsStream = nd.requestEvents(requestMessage, retryRelay);
+            await for (final retryEvent in retryEventsStream) {
+              if (streamController.isClosed) break;
+              if (retryEvent.indicatesError) {
+                dislikedRelaysUrls.add(retryRelay.url);
+                throw Exception(RelayRequestFailedException());
```

Comment:
It's enough to throw only the `RelayRequestFailedException`. There is no need to wrap additionally into `Exception`

Reply:
of course, fixed